File: Backend/Fesnuk.API/Models/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Fesnuk.API.Models
{
    public class Post
    {
        [Key]
        public Guid PostId { get; set; } = Guid.NewGuid();
        [Required]
        public Guid UserId { get; set; }
        [Required]
        public string MediaUrl { get; set; }
        [MaxLength(2200)]
        public string? Caption { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [ForeignKey("UserId")]
        public virtual User User { get; set; }
    }
}

File: Frontend/frontend-next/src/types/post.ts
// Match with Backend DTOs

export interface PostAuthor
{
    userId: string;
    username: string;
    userProfilePictureUrl?: string;
}

export interface Post
{
    postId: string;
    mediaUrl: string;
    caption?: string;
    createdAt: string;
    user: PostAuthor;
}

export interface CreatePostData
{
    mediaFile: File;
    caption?: string;
}